pkg/storage/unified/client.go

```diff
@@ -153,3 +161,74 @@ func newResourceClient(conn *grpc.ClientConn, cfg *setting.Cfg, features feature
 	}
 	return resource.NewRemoteResourceClient(tracer, conn, clientCfgMapping(grpcutils.ReadGrpcClientConfig(cfg)), cfg.Env == setting.Dev)
 }
+
+// grpcConn creates a new gRPC connection to the provided address.
+func grpcConn(address string, reg prometheus.Registerer) (*grpc.ClientConn, error) {
+	// This works for now as the Provide function is only called once during startup.
+	// We might eventually want to tight this factory to a struct for more runtime control.
+	metrics := ClientMetrics{
+		requestDuration: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "resource_server_client_request_duration_seconds",
+			Help:    "Time spent executing requests to the resource server.",
+			Buckets: prometheus.ExponentialBuckets(0.008, 4, 7),
+		}, []string{"operation", "status_code"}),
+	}
+
+	// Report gRPC status code errors as labels.
+	var instrumentationOptions []middleware.InstrumentationOption
+	instrumentationOptions = append(instrumentationOptions, middleware.ReportGRPCStatusOption)
+	unary, stream := grpcclient.Instrument(metrics.requestDuration, instrumentationOptions...)
```

Comment:
It's this call that adds `ClientUserHeaderInterceptor` interceptor, which takes tenant from the context, and adds it to the outgoing metadata.

I'd suggest to inline this call and simply skip adding `ClientUserHeaderInterceptor`. Then we don't need to add fake user to the context in yet-another interceptor.

Reply:
Done 👍